<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Visualizer</class>
 <widget class="QMainWindow" name="Visualizer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>781</width>
    <height>595</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Sorting Visualizer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGraphicsView" name="visualizerGraphicsView">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>750</width>
      <height>370</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(30, 30, 30);</string>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="interactive">
     <bool>false</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignBottom|Qt::AlignRight|Qt::AlignTrailing</set>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>390</y>
      <width>211</width>
      <height>171</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label_TSC">
       <property name="font">
        <font>
         <pointsize>15</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Time and Space Complexity</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="algorithmType_label">
       <property name="text">
        <string>Sorting Algorithm: Bubble sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="bestCase_label">
       <property name="text">
        <string>Best Case: O(n²)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="averageCase_label">
       <property name="text">
        <string>Average: O(n²)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="worstCase_label">
       <property name="text">
        <string>Worst Case: O(n²)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="spaceO_label">
       <property name="text">
        <string>Space Complexity: O(1)</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_speed">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>450</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Speed:</string>
    </property>
   </widget>
   <widget class="QSlider" name="sortingSpeed_slider">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>470</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>25</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
    <property name="sliderPosition">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>397</y>
      <width>241</width>
      <height>151</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_barColor">
         <property name="font">
          <font>
           <pointsize>15</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Bar Color:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="barColor_comboBox">
         <property name="font">
          <font>
           <pointsize>13</pointsize>
          </font>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Green</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Red</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Blue</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>White</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Pink</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label_poiinterColor">
         <property name="font">
          <font>
           <pointsize>15</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Pointer Color:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="pointerColor_comboBox">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Red</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Green</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Blue</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>White</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Pink</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="label_visBG">
         <property name="font">
          <font>
           <pointsize>15</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Visualizer Color:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="backgroundColor_comboBox">
         <item>
          <property name="text">
           <string>Dark Gray</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>White</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Black</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="timeTaken_label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>500</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>17</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Time Taken (No Delay): </string>
    </property>
   </widget>
   <widget class="QPushButton" name="createArray_button">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>420</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>New Array</string>
    </property>
   </widget>
   <widget class="QPushButton" name="sortButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>380</y>
      <width>130</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Sort</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pause_button">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>460</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Pause</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_selectAlgorithm">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>390</y>
      <width>125</width>
      <height>19</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Select Algorithm:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="algorithmType_comboBox">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>410</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <item>
     <property name="text">
      <string>Bubble Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Selection Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Insertion Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Quick Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Merge Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Radix Sort</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="forceQuit_button">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>530</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Force Quit</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>781</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
